exercise/is-winner.js

```diff
@@ -74,66 +76,74 @@ const winners = () => {
       if (!match.length) throw Error('Results Not Found')
       return match
     },
+    addCountry: (country) => countries.push(country),
+    addResults: (result) => results.push(result),
   }
-}
+})()
 /// * //
 
 /// * //
 export const tests = []
 const t = (f) => tests.push(f)
 
-//testing correct continent but wrong number of times
-t(({ eq }) =>
-  isWinner('England').then((data) => {
-    return eq(
-      data,
-      'England is not what we are looking for because of the number of times it was champion'
-    )
-  })
+// testing correct continent but wrong number of times
+t(async ({ eq }) =>
+  eq(
+    await isWinner('England'),
+    'England is not what we are looking for because of the number of times it was champion'
+  )
 )
 
-//testing non winner country
-t(({ eq }) =>
-  isWinner('Colombia').then((data) => {
-    return eq(data, 'Colombia never was a winner')
-  })
+// testing non winner country
+t(async ({ eq }) =>
+  eq(await isWinner('Colombia'), 'Colombia never was a winner')
 )
 
-//testing wrong continent country
-t(({ eq }) =>
-  isWinner('Uruguay').then((data) => {
-    return eq(
-      data,
-      'Uruguay is not what we are looking for because of the continent'
-    )
-  })
+// testing wrong continent country
+t(async ({ eq }) =>
+  eq(
+    await isWinner('Uruguay'),
+    'Uruguay is not what we are looking for because of the continent'
+  )
 )
 
-//testing no country
-t(({ eq }) =>
-  isWinner('').then((data) => {
-    return eq(data, ' never was a winner')
-  })
+// testing no country
+t(async ({ eq }) => eq(await isWinner(''), ' never was a winner'))
+
+// testing correct number of times but wrong continent
+t(async ({ eq }) =>
+  eq(
+    await isWinner('Brazil'),
+    'Brazil is not what we are looking for because of the continent'
+  )
 )
 
-//testing correct number of times but wrong continent
-t(({ eq }) =>
-  isWinner('Brazil').then((data) => {
-    return eq(
-      data,
-      'Brazil is not what we are looking for because of the continent'
-    )
-  })
+// testing correct number of times and correct continent
+t(async ({ eq }) =>
+  eq(
+    await isWinner('Germany'),
+    'Germany won the FIFA World Cup in 1954, 1974, 1990, 2014 winning by 3-2, 2-1, 1-0, 1-0'
+  )
 )
 
-//testing correct number of times and correct continent
-t(({ eq }) =>
-  isWinner('Germany').then((data) => {
-    return eq(
-      data,
-      'Germany won the FIFA World Cup in 1954, 1974, 1990, 2014 winning by 3-2, 2-1, 1-0, 1-0'
-    )
-  })
+// testing correct number of times and correct continent, for the fake country
+t(async ({ eq, ctx }) =>
+  eq(
+    await isWinner(ctx.name),
+    `${ctx.name} won the FIFA World Cup in 2022, 2026, 2030 winning by 1-0, 3-1, 2-1`
+  )
 )
 
 Object.freeze(tests)
+
+export const setup = () => {
```

Comment:
create a random country with fake results to ensure they don't just copy / paste the answers